### ISSUE TITLE
TO DO: Update less-preview with all releases of Less from 2.5.3 to 3.0

### ISSUE THREAD

**Issue by matthew-dean** [less/less.js#3194]
Currently, there's not an easy way to demo changes in code output online between versions. Would someone be willing to take the time to update http://lesscss.org/less-preview ?

The UI (dropdown) could use a refresh as well, but that's minor.

**Comment by seven-phases-max** [comment]
It's not so easy (e.g. https://github.com/less/less-preview/issues/6) :(. At some point I'm afraid someone has to come up with a much more light-weighted implementation written from scratch probably (after all it's nothing but code-mirror editor + less.js).

**Comment by rejas** [comment]
I took a quick look at the code of the less-preview site and to me it seems that it would autoamtically show all versions of less that it gets from the npm registry. So the real issue would be to deploy it which should take care of the missing versions, am I correct?

**Comment by rejas** [comment]
Sorry, @seven-phases-max , I dont get your meaning. Could you rephrase that?

**Comment by rejas** [comment]
Well, @matthew-dean, then the question to you: who has the rights / administrative capabilities to deploy a new version of it to the site?

**Comment by seven-phases-max** [comment]
We both have.

**Comment by matthew-dean** [comment]
@rejas - See: https://github.com/orgs/less/people

**Comment by matthew-dean** [comment]
Okay, I got the `less-preview` build working locally, but I can't for the life of me figure out how to publish the updated build to `gh-pages`. The `gulp` commands seem to bundle the correct output for `gh-pages`, but the files don't seem to be updated as far as `git` is concerned, so I'm very confused.

It's frustrating because if you pull https://github.com/less/less-preview/ and do `npm install` and then `gulp watch`, you can see it working. But from there, I'm stuck.  :/

> At some point I'm afraid someone has to come up with a much more light-weighted implementation written from scratch probably (after all it's nothing but code-mirror editor invoking less.js render).

Yeah I'm not sure why it's so over-engineered. But if someone can get it to deploy, it otherwise works now.

**Comment by matthew-dean** [comment]
It looks like I may have broken preview, not sure....

**Comment by matthew-dean** [comment]
### Update

Okay, the previous site was just an incredible mess of dependencies, all of which were failing at different points. I rebuilt Less-preview as just a simple, straight-up Nuxt.js app with one page with two Ace editors.

I didn't rebuild the option to select previous Less versions. If someone else wants to take that on, be my guest. It's live at http://lesscss.org/less-preview/

**Comment by kevinramharak** [comment]
It seems that the current version is `3.10.3`? So this issue is now a feature request to be able to select previous less versions?

**Comment by matthew-dean** [comment]
@kevinramharak Someone had built a version originally where the versions of Less was selectable. However, the way it was built was not maintainable. I rebuilt it with just a simple preview, with the latest version. So yes, if someone wanted to do the work of creating a dropdown with recent X versions, that would be great.

**Comment by lneugebauer** [comment]
@matthew-dean if I understand you correctly this feature has been implemented in the meantime and this issue can be closed?
At least I'm able to select different version on the current preview page and it had an update to vue3 not too long ago.

**Comment by iChenLei** [comment]
closed by https://github.com/less/less-preview/pull/11
